impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -120,6 +124,12 @@ public int compare(ValueHolder<?> t, ValueHolder<?> u) {
   private final Class<V> valueType;
   private final Copier<K> keyCopier;
   private final Copier<V> valueCopier;
+  
+  private final SizeOfEngine sizeOfEngine;
+  private final long faultOffset;
+  
+  private final boolean byteSized;
```

Comment:
I'm not fond of that variable and all those if (byteSized) {} blocks. Couldn't we inject a no-op sizeof engine that always return 0 as the size when a store is configured with max entries?